File: src/IsbInterop/Data/IRequisite.vb
Imports IsbInterop.DataTypes.Enumerable

Namespace Data

  ''' <summary>
  ''' Реквизит.
  ''' </summary>
  Public Interface IRequisite
    Inherits IBaseIsbObject

    ''' <summary>
    ''' Целочисленное значение реквизита.
    ''' </summary>
    ReadOnly Property AsInteger As Integer

    ''' <summary>
    ''' Строковое значение реквизита.
    ''' </summary>
    ReadOnly Property AsString As String

    ''' <summary>
    ''' Значение реквизита в качестве даты.
    ''' </summary>
    ReadOnly Property AsDate As DateTime

    ''' <summary>
    ''' Тип реквизита.
    ''' </summary>
    ReadOnly Property DataType As TReqDataType

    ''' <summary>
    ''' Имя поля реквизита в базе.
    ''' </summary>
    ReadOnly Property FieldName As String

    ''' <summary>
    ''' Признак, указывающий, что реквизит равен null.
    ''' </summary>
    ReadOnly Property IsNull As Boolean

    ''' <summary>
    ''' Имя реквизита.
    ''' </summary>
    ReadOnly Property Name As String

    ''' <summary>
    ''' Значение реквизита.
    ''' </summary>
    Property Value As Object
  End Interface
End Namespace

File: src/IsbInterop/Data/Requisites/IRequisite.cs
using IsbInterop.DataTypes.Enumerable;
using System;

namespace IsbInterop.Data.Requisites
{
  /// <summary>
  /// Реквизит.
  /// </summary>
  public interface IRequisite : IBaseIsbObject
  {
    /// <summary>
    /// Целочисленное значение реквизита.
    /// </summary>
    int AsInteger { get; }

    /// <summary>
    /// Строковое значение реквизита.
    /// </summary>
    string AsString { get; }

    /// <summary>
    /// Значение реквизита в качестве даты.
    /// </summary>
    DateTime AsDate { get; }

    /// <summary>
    /// Тип реквизита.
    /// </summary>
    TReqDataType DataType { get; }

    /// <summary>
    /// Имя поля реквизита в базе.
    /// </summary>
    string FieldName { get; }

    /// <summary>
    /// Имя реквизита.
    /// </summary>
    string Name { get; }

    /// <summary>
    /// Признак, указывающий, что реквизит равен null.
    /// </summary>
    bool IsNull { get; }

    /// <summary>
    /// Значение реквизита.
    /// </summary>
    object Value { get; set; }
  }
}
